Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp

```diff
@@ -1084,30 +1084,30 @@ namespace AZ::Render
     // Find the trajectory path for a given actor instance
     AtomActorDebugDraw::TrajectoryTracePath* AtomActorDebugDraw::FindTrajectoryPath(const EMotionFX::ActorInstance* actorInstance)
     {
-        for (TrajectoryTracePath* trajectoryPath : m_trajectoryTracePaths)
+        for (const auto& trajectoryPath : m_trajectoryTracePaths)
         {
             if (trajectoryPath->m_actorInstance == actorInstance)
             {
-                return trajectoryPath;
+                return trajectoryPath.get();
             }
         }
 
         // We haven't created a path for the given actor instance yet, do so
-        TrajectoryTracePath* tracePath = new TrajectoryTracePath();
+        AZStd::unique_ptr<TrajectoryTracePath> tracePath = AZStd::make_unique<TrajectoryTracePath>();
 
         tracePath->m_actorInstance = actorInstance;
         tracePath->m_traceParticles.reserve(512);
 
-        m_trajectoryTracePaths.emplace_back(tracePath);
-        return tracePath;
+        m_trajectoryTracePaths.emplace_back(AZStd::move(tracePath));
+        return tracePath.get();
```

Comment:
This uses the `tracePath` variable after it has been moved from, so you don't know exactly what the state of that variable will be. Because `emplace_back` returns a reference to the inserted object, it would be better to write this instead:
```suggestion
        return m_trajectoryTracePaths.emplace_back(AZStd::move(tracePath)).get();
```

Reply:
done.